app/code/Magento/MediaContentSynchronization/Test/Integration/Model/SynchronizeIdentitiesTest.php

```diff
@@ -0,0 +1,159 @@
+     */
+    private $getAssetIds;
+
+    /**
+     * @var GetContentByAssetIdsInterface
+     */
+    private $getContentIdentities;
+
+    /**
+     * @var SynchronizeIdentitiesInterface
+     */
+    private $synchronizeIdentities;
+
+    /**
+     * @var SynchronizeInterface
+     */
+    private $synchronize;
+
+    protected function setUp(): void
+    {
+        $this->contentIdentityFactory = Bootstrap::getObjectManager()->get(ContentIdentityInterfaceFactory::class);
+        $this->getAssetIds = Bootstrap::getObjectManager()->get(GetAssetIdsByContentIdentityInterface::class);
+        $this->synchronizeIdentities = Bootstrap::getObjectManager()->get(SynchronizeIdentitiesInterface::class);
+        $this->synchronize = Bootstrap::getObjectManager()->get(SynchronizeInterface::class);
+        $this->getContentIdentities = Bootstrap::getObjectManager()->get(GetContentByAssetIdsInterface::class);
+    }
+
+    /**
+     * @dataProvider filesProvider
+     * @magentoDataFixture Magento/MediaContentCatalog/_files/category_with_asset.php
+     * @magentoDataFixture Magento/MediaContentCatalog/_files/product_with_asset.php
+     * @magentoDataFixture Magento/MediaContentCms/_files/page_with_asset.php
+     * @magentoDataFixture Magento/MediaContentCms/_files/block_with_asset.php
+     * @magentoDataFixture Magento/MediaGallery/_files/media_asset.php
+     * @param array $mediaContentIdentities
+     * @throws IntegrationException
+     */
+    public function testExecute(array $mediaContentIdentities): void
+    {
+        $this->assertNotEmpty($mediaContentIdentities);
+        $this->synchronizeIdentities->execute($mediaContentIdentities);
+
+        foreach ($mediaContentIdentities as $contentIdentity) {
+            $assetId = 2020;
+            $identity = $this->contentIdentityFactory->create($contentIdentity);
+            $this->assertEquals([$assetId], $this->getAssetIds->execute($identity));
+
+            $synchronizedContentIdentities = $this->getContentIdentities->execute([$assetId]);
+            $this->assertEquals(4, count($synchronizedContentIdentities));
+
+            $syncedIds = [];
+            foreach ($synchronizedContentIdentities as $syncedContentIdentity) {
+                $syncedIds[] = (int)$syncedContentIdentity->getEntityId();
+            }
+            $this->assertContains($contentIdentity['entityId'], $syncedIds);
+        }
+    }
+
+    /**
+     * Data provider
+     *
+     * @return array
+     */
+    public function filesProvider(): array
+    {
+        return [
+            [
+                [
+                    $this->getCategoryIdentities(),
+                    $this->getProductIdentities(),
+                    $this->getCmsPageIdentities(),
+                    $this->getCmsBlockIdentities()
+                ]
+            ]
+        ];
+    }
+
+    /**
+     * Format category media content identities
+     */
+    public function getCategoryIdentities()
```

Comment:
I don't think these functions are necessary, please include the data directly in the dataprovider function